### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'path'

### ISSUE THREAD

**Issue by lamres** [automl/auto-sklearn#1149]
Hi.

## Describe the bug ##
I get this error:
`Exception ignored in: <function AutoML.__del__ at 0x7fc570a90430>
Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.8/site-packages/autosklearn/automl.py", line 1564, in __del__
  File "/opt/anaconda3/lib/python3.8/site-packages/autosklearn/util/backend.py", line 155, in delete_directories
  File "/opt/anaconda3/lib/python3.8/site-packages/autosklearn/util/backend.py", line 108, in temporary_directory
AttributeError: 'NoneType' object has no attribute 'path'`

## To Reproduce ##
1. Load the model from a pickle.
2. Make a prediction. The prediction is ok.
3. This error occurs in the end of code.

## Installation ##

Python 3.8 on MacOS, the version of the auto-sklearn is 0.12.6

**Comment by franchuterivera** [comment]
Hello, thanks for reporting. 

If I take a simple classification example like [this](https://automl.github.io/auto-sklearn/master/examples/20_basic/example_classification.html#sphx-glr-examples-20-basic-example-classification-py) and add this line at the end:
```
  import pickle
  import gc
   with open('estimator.pickle', 'wb') as handle:
       pickle.dump(automl, handle, protocol=pickle.HIGHEST_PROTOCOL)
   
   del automl
   gc.collect()
   
   with open('estimator.pickle', 'rb') as handle:
       b = pickle.load(handle)
   del b
   gc.collect()

```
I am not able to reproduce. Furthermore, if I then open a python session and do:
```
>>> import pickle
>>> with open('estimator.pickle', 'rb') as handle:
...     b = pickle.load(handle)
>>> b.predict(<input>)
>>>exit()
```
How is your use case different? I do not know if this problem in your case is specific to MacOS. Is it possible to reproduce this on a virtual machine with for example Ubuntu?

**Comment by DankeyDunk** [comment]
I'm receiving the same error message as well:

```
Exception ignored in: <function AutoML.__del__ at 0x2b4791023ca0>
Traceback (most recent call last):
  File "/deac/generalGrp/khuriGrp/software/auto-sklearn/lib/python3.8/site-packages/autosklearn/automl.py", line 1564, in __del__
  File "/deac/generalGrp/khuriGrp/software/auto-sklearn/lib/python3.8/site-packages/autosklearn/util/backend.py", line 121, in delete_directories
  File "/deac/generalGrp/khuriGrp/software/auto-sklearn/lib/python3.8/site-packages/autosklearn/util/backend.py", line 101, in output_directory
AttributeError: 'NoneType' object has no attribute 'path'
```

Currently running Python 3.8.10 with auto-sklearn version 0.12.6

I was able to produce this error with the simple classification example [linked](https://automl.github.io/auto-sklearn/master/examples/20_basic/example_classification.html#sphx-glr-examples-20-basic-example-classification-py) above.

I am running on RHEL7 in a Python virtual environment.

**Comment by franchuterivera** [comment]
Thanks a lot for reporting. I was able to reproduce this problem on red hat, and I am proposing a PR that will be available on the next release to solve this problem.

**Comment by eddiebergman** [comment]
This was addressed with PR #1155 and is now in the development branch. Hopefully this fixes the error for now so I will close the issue but if the issue arises again, please feel free to comment here and I will re-open it!
